Pipewire as a JACK Replacement (#480)
getsolus/help-center-docs: docs/user/software/sound/index.md

# Sound

Solus comes with Pipewire configured by default.

## Pipewire as a JACK Replacement

> 

Reviewer: Please add a sentence or two explaining why someone might want to do this

Firstly, install `pipewire-jack`

```bash
sudo eopkg it pipewire-jack
```

> Currently, using `pipewire-jack` as a `JACK` replacement is not as smooth as one would think. One must get their hands dirty a little to get JACK apps to behave appropriately 🛠

Reviewer: We don't use emojis this way. Please remove 🛠

### Issues to be fixed

> With no configuration pipewire will not be able to switch to the appropriate sample-rate for the project.

Reviewer: put pipewire in backticks, since you are referring to a package

![Pipewire Rates Issue](ardour-pipewire-rates-issue.png)

### Configuration

Fortunately both the `pipewire` and `pipewire-jack` packages come with the configs we need.

> Open up your terminal of choice and input the commands below

Reviewer: Add a sentence here explaining what the string of commands will do, like: "The following commands copy the default configurations to your user configuration directory, where you can modify them."

```bash
cd .config/
mkdir -p pipewire/pipewire.conf.d/
sudo cp /usr/share/pipewire/pipewire.conf.avail/10-rates.conf pipewire/pipewire.conf.d/
sudo cp /usr/share/pipewire/jack.conf pipewire/

> sudo chown yourUserName:yourUserName pipewire/jack.conf pipewire/pipewire.conf.d/10-rates.conf

Reviewer: Can't you get the same result, without `chown`, if you just use `cp` rather than `sudo cp` ?

```

Now open the `jack.conf` in your text editor and

1. Search for `node.rate`
2. Uncomment `node.rate`
3. Input the sample-rate you wish JACK applications to open with after the `1/`

>    - common rates: 44100 48000 88200 96000

Reviewer: You should probably suggest a good default rate

**Without this `Pipewire` wont dynamically adjust the sample-rate for you whenever a `JACK` application opens**

![jack.conf change gif](config-change.gif)

Now reboot your system.

:::note
After applying the configuration above there **WILL** be a delay of up to 2 seconds when beginning to play audio. This is due to the 10-rates.conf that allows `Pipewire` to dynamically change sample-rates. Although, once the sample-rate is set everything is instant.
:::

### Not Done Yet

> Lastly, we need to enable the `Pro Audio` sound configuration for our card. Not doing so will result in terrible Xruns (pops and crackles) even with all applied configuration.

Reviewer: What are Xruns?

Reviewer 2: https://alsa.opensrc.org/Xruns

> **On Budgie**: `pavucontrol` is required to get access to the `Pro Audio` config. So install it

Reviewer: Presumably, the Budgie changes also work for GNOME and XFCE. Please note that

```bash
sudo eopkg it pavucontrol
```

> Now open the program, switch over to the `Configuration` tab and select the `Pro Audio` option from the dropdown menu for your card

Reviewer 3: Use bolded text when you refer to the UI.

```suggestion
Open `pavucontrol`, go to the **Configuration** tab and select the **Pro Audio** option from the dropdown menu for your card.
```

![Pavucontrol Config Change](pavu-config.gif)

> **On KDE**: everything you need is in the `Audio` section of system settings. Simply choose from the dropdown same as above.

Reviewer 3: Avoid saying "same as above". Always include the full instructions.
```suggestion
**On KDE**: Go to the `Audio` section of system settings. Select the **Pro Audio** option from the dropdown menu for your card.

> **All is done! Happy audio production!!** 🥰

Reviewer 3: Remove the emoji and the bolded text.

```suggestion
All is done! Happy audio production!!
```

> ### Limits

Reviewer 3: I think troubleshooting is a more appropriate title

> If this is annoying you

Reviewer 3: Try to use a more neutral language.

```suggestion
If you see the following error message:
```

> ![Ardour yelling about limited amount of locked memory](ulimits.png)

Reviewer 3: Same here. Use neutral language
```suggestion
![Ardour warning about the limited amount of locked memory](ulimits.png)
```

You can adjust it by configuring the `limits.conf`. In this case we'll be using the `limits.d` for simplicity.

> Input the commands below

Reviewer 3: As mentioned before, explain what the commands will do.

```suggestion
Execute the commands below:
```
